test/mpi/maint/gentests_dtp.sh

```diff
@@ -190,6 +190,13 @@ while read -r line ; do
                     echo "${testname} $procs arg=-type=${type} arg=-count=${sendcount} arg=-seed=$seed arg=-testsize=${gputestsize} ${other_args} arg=-origmem=device arg=-targetmem=device $timelimit" >> ${builddir}/${testdir}/testlist.gpu
                     seed=$((seed + 1))
                     fi
+            elif [ $testdir = "coll" ] ; then
+                echo "${testname} $procs arg=-type=${type} arg=-count=${sendcount} arg=-seed=$seed arg=-testsize=${testsize} ${other_args} arg=-memtype=host $timelimit" >> ${builddir}/${testdir}/testlist.dtp
+                seed=$((seed + 1))
+                echo "${testname} $procs arg=-type=${type} arg=-count=${sendcount} arg=-seed=$seed arg=-testsize=${gputestsize} ${other_args} arg=-memtype=reg_host $timelimit" >> ${builddir}/${testdir}/testlist.gpu
+                seed=$((seed + 1))
+                echo "${testname} $procs arg=-type=${type} arg=-count=${sendcount} arg=-seed=$seed arg=-testsize=${gputestsize} ${xother_args} arg=-memtype=device $timelimit" >> ${builddir}/${testdir}/testlist.gpu
+                seed=$((seed + 1))
```

Comment:
For the collectives, do we not want to generate cases where some processes use one type of memory and some processes don't?  How about modifying the tests so the memtype argument is used as "memtype + rank" by each rank?

Reply:
I will add this and re-submit.